### PR TITLE
Add x11 service for unclutter-xfixes

### PR DESCRIPTION
**Pull request by erikryb**
###### Motivation for this change

This is a service similar to the unclutter service, but for the new unclutter-xfixes. There is some [effort](https://github.com/Airblader/unclutter-xfixes/issues/14) to make unclutter-xfixes compatible with unclutter, but for now it is probably best to keep them as two different modules.
###### Things done
- [x] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
  - [x] NixOS
  - [ ] OS X
  - [ ] Linux
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: nixos/modules/services/x11/unclutter-xfixes.nix [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+      description = "unclutter-xfixes derivation to use.";
+    };
+
+    timeout = mkOption {
+      description = "Number of seconds before the cursor is marked inactive";
+      type = types.int;
+      default = 1;
+    };
+
+    threeshold = mkOption {
```
Comment (bjornfor) [inline nixos/modules/services/x11/unclutter-xfixes.nix]: Typo (one 'e' too many).

File: nixos/modules/services/x11/unclutter-xfixes.nix [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+      description = "Minimum number of pixels considered cursor movement";
```
Comment (bjornfor) [inline nixos/modules/services/x11/unclutter-xfixes.nix]: Descriptions should end with a dot (full sentences).

File: nixos/modules/services/x11/unclutter-xfixes.nix [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+      type = types.listOf types.str;
+      default = [];
+      example = [ "exclude-root" "ignore-scrolling" "fork" ];
+    };
+  };
+
+  config = mkIf cfg.enable {
+    systemd.user.services.unclutter-xfixes = {
+      description = "unclutter-xfixes";
+      wantedBy = [ "default.target" ];
```
Comment (groxxda) [inline nixos/modules/services/x11/unclutter-xfixes.nix]: Minor nitpick: Imo this should be `graphical.target`.

Comment (erikryb) [inline nixos/modules/services/x11/unclutter-xfixes.nix]: Yeah, that makes sense. Better to be explicit.

**Comment by groxxda** [comment]
We also might want some ordering `After=display-manager.service` and possibly `Requisite=display-manager.service` (?).

**Comment by erikryb** [comment]
Why "Requisite" instead of "Requires"?

**Comment by groxxda** [comment]
Well, in my opinion if `display-manager` is not pulled in on it's own, `unclutter` should respect that and just refuse to start.. `PartOf` or `BindsTo` could also be valid choices, that's why I added a :question: :wink:

**Comment by erikryb** [comment]
I see. It seems that `BindsTo` is a stronger version of `Requires`, so 
that should also not be used for the same reason. Am I correct in that 
`PartOf` is a weaker version of `Requisite`, that is, does `Requisite` 
also restart or stop `unclutter` when `display-manager` restarts or 
stops? If so, I believe `Requisite` is the best option, as you first 
stated.

**Comment by groxxda** [comment]
I did not try the exact behavior, but `Requisite` sounded appropriate to me. I expect `clutter` to fail anyway when `display-manager` stops / fails / restarts, so it's not really that important whether it's propagated by systemd. But feel free to try and report what happens :+1: 

Apart from this, if you look at the `redshift` service that's quite similar, it will get merged even if it's not perfect.. :disappointed:

**Comment by bjornfor** [comment]
Why did you close?

**Comment by erikryb** [comment]
It was a mistake. I rebased my master branch and the pull request got closed. I guess that's why I should have used a dedicated branch for the pull request. How do I reopen it? Or should I just start a new pull request?

**Comment by joachifm** [comment]
@erikryb I rebased & merged this at 2f0cc0d3f055d64ca3f4c8664c12e8f02a833d0d. Thank you

**Comment by erikryb** [comment]
Thanks!

**Pull request closed without merging**